File: enggo-backend/src/main/java/com/nhom12/enggo_backend/dto/response/gamification/InviteResponse.java
package com.nhom12.enggo_backend.dto.response.gamification;

import com.nhom12.enggo_backend.dto.request.exam.RandomBlueprintRequest;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class InviteResponse {
    int inviteId;
    Integer inviterPlayerId;
    Integer inviteePlayerId;
    String inviterUsername;
    String inviteeUsername;
    RandomBlueprintRequest randomBlueprintRequest;
    String status; // PENDING, ACCEPTED, DECLINED, TIMEOUT
}

File: AppEngGo/app/src/main/java/com/example/appenggo/model/InviteResponse.kt
package com.example.appenggo.model

import com.google.gson.annotations.SerializedName

/**
 * Mirrors backend InviteResponse DTO.
 */
 data class InviteResponse(
    @SerializedName("inviteId")
    val inviteId: Int,

    @SerializedName("inviterPlayerId")
    val inviterPlayerId: Int?,

    @SerializedName("inviteePlayerId")
    val inviteePlayerId: Int?,

    @SerializedName("inviterUsername")
    val inviterUsername: String?,

    @SerializedName("inviteeUsername")
    val inviteeUsername: String?,

    @SerializedName("randomBlueprintRequest")
    val randomBlueprintRequest: RandomBlueprintRequest?,

    @SerializedName("status")
    val status: String?
)
